google/cloud/storage/client.h

```diff
@@ -155,6 +155,28 @@ class Client {
     return ObjectReadStream(raw_client_, request);
   }
 
+  /**
+   * Delete an object.
+   *
+   * @param bucket_name the name of the bucket that contains the object.
+   * @param object_name the name of the object to be deleted.
+   * @param parameters a variadic list of optional parameters. Valid types for
+   *   this operation include
+   *   `Generation`, `IfGenerationMatch` / `IfGenerationNotMatch`,
+   *   `IfMetagenerationMatch` / `IfMetagenerationNotMatch`, and `UserProject`.
```

Comment:
Why do we include a UserProject command? The project is determined by the bucket, so it shouldn't be necessary to include it explicitly, and including it explicitly opens the possibility for a mismatch between the specified project and the project the bucket actually belongs to. Also, we don't include project in any of the GCS delete object APIs.

Reply:
I am just going by the documentation:

https://cloud.google.com/storage/docs/json_api/v1/objects/delete

where it says you need to use `userProject` for Requester Pays buckets.

Reply:
Ah, right, I forgot about Requester Pays buckets.